src/main/java/org/kiwiproject/beta/reflect/KiwiReflection2.java

```diff
@@ -160,4 +167,40 @@
     public static boolean isPackagePrivate(Class<?> clazz) {
         return !isPublic(clazz) && !isProtected(clazz) && !isPrivate(clazz);
     }
+
+    /**
+     * Get the type information for the given {@link Field}.
+     *
+     * @param field the Field to check
+     * @return the type information
+     */
+    public static TypeInfo typeInformationOf(@NonNull Field field) {
+        checkArgumentNotNull(field, "field to inspect must not be null");
+        var genericType = field.getGenericType();
+        return typeInformationOf(genericType);
+    }
+
+    /**
+     * Get the type information for the given {@link Type}.
+     * <p>
+     * This is a convenience method that delegates to {@link TypeInfo#ofType(Type)}.
+     *
```

Comment:
CodeQL is correct here, and maybe the better thing to do is simply remove the methods that don't do anything other than call `TypeInfo` methods. Then there is only one method in `KiwiReflection2` which accepts a `Field`. And perhaps that should be renamed as `typeInformationOfField`. Thoughts @dsingley ?

Reply:
Actually if I remove the method that accepts `ParameterizedType` I think that will be OK, since `Field` and `ParameterizedType` are "substantially different" whereas `ParameterizedType` is a subtype of `Type` and is _not_ substantially different. And since `TypeInfo#ofType` accepts `Type` and internally makes sure to handle it as a simple type or as a parameterized type, I think this is the better way.